src/modules/robot/Robot.cpp

```diff
@@ -195,20 +201,20 @@ void Robot::on_config_reload(void *argument)
     Pin gamma_dir_pin;
     Pin gamma_en_pin;
 
-    alpha_step_pin.from_string( THEKERNEL->config->value(alpha_step_pin_checksum )->by_default("2.0"  )->as_string())->as_output();
-    alpha_dir_pin.from_string(  THEKERNEL->config->value(alpha_dir_pin_checksum  )->by_default("0.5"  )->as_string())->as_output();
-    alpha_en_pin.from_string(   THEKERNEL->config->value(alpha_en_pin_checksum   )->by_default("0.4"  )->as_string())->as_output();
-    beta_step_pin.from_string(  THEKERNEL->config->value(beta_step_pin_checksum  )->by_default("2.1"  )->as_string())->as_output();
-    beta_dir_pin.from_string(   THEKERNEL->config->value(beta_dir_pin_checksum   )->by_default("0.11" )->as_string())->as_output();
-    beta_en_pin.from_string(    THEKERNEL->config->value(beta_en_pin_checksum    )->by_default("0.10" )->as_string())->as_output();
-    gamma_step_pin.from_string( THEKERNEL->config->value(gamma_step_pin_checksum )->by_default("2.2"  )->as_string())->as_output();
-    gamma_dir_pin.from_string(  THEKERNEL->config->value(gamma_dir_pin_checksum  )->by_default("0.20" )->as_string())->as_output();
-    gamma_en_pin.from_string(   THEKERNEL->config->value(gamma_en_pin_checksum   )->by_default("0.19" )->as_string())->as_output();
+    alpha_step_pin.from_string( THEKERNEL->config->value((unsigned int)alpha_step_pin_checksum )->by_default("2.0"  )->as_string())->as_output();
```

Comment:
there should always be a valid answer, there is no case where it is ok to not return something.

you would need to move further up the chain to stop a print if an invalid gcode or position were 
given and there currently is no way to detect invalid gcode.


On 06/10/2015 04:35 PM, Douglas Pearless wrote:
> In src/modules/robot/Robot.cpp
> <https://github.com/Smoothieware/Smoothieware/pull/673#discussion_r32177908>:
>
>> @@ -195,20 +201,20 @@ void Robot::on_config_reload(void *argument)
>>      Pin gamma_dir_pin;
>>      Pin gamma_en_pin;
>>
>> -    alpha_step_pin.from_string( THEKERNEL->config->value(alpha_step_pin_checksum )->by_default("2.0"  )->as_string())->as_output();
>> -    alpha_dir_pin.from_string(  THEKERNEL->config->value(alpha_dir_pin_checksum  )->by_default("0.5"  )->as_string())->as_output();
>> -    alpha_en_pin.from_string(   THEKERNEL->config->value(alpha_en_pin_checksum   )->by_default("0.4"  )->as_string())->as_output();
>> -    beta_step_pin.from_string(  THEKERNEL->config->value(beta_step_pin_checksum  )->by_default("2.1"  )->as_string())->as_output();
>> -    beta_dir_pin.from_string(   THEKERNEL->config->value(beta_dir_pin_checksum   )->by_default("0.11" )->as_string())->as_output();
>> -    beta_en_pin.from_string(    THEKERNEL->config->value(beta_en_pin_checksum    )->by_default("0.10" )->as_string())->as_output();
>> -    gamma_step_pin.from_string( THEKERNEL->config->value(gamma_step_pin_checksum )->by_default("2.2"  )->as_string())->as_output();
>> -    gamma_dir_pin.from_string(  THEKERNEL->config->value(gamma_dir_pin_checksum  )->by_default("0.20" )->as_string())->as_output();
>> -    gamma_en_pin.from_string(   THEKERNEL->config->value(gamma_en_pin_checksum   )->by_default("0.19" )->as_string())->as_output();
>> +    alpha_step_pin.from_string( THEKERNEL->config->value((unsigned int)alpha_step_pin_checksum )->by_default("2.0"  )->as_string())->as_output();
>
> Hmm,
> â€¦
> On re-reading the Robot.cpp I realise that all methods except "void
> Robot::reset_position_from_current_actuator_position()" pass actuator_pos as an unitialised array
> (which depending on the compiler may initialise as [0,0,0] ). There does not appear to be a way that
> the void cartesian_to_actuator( float[], float[] ); void actuator_to_cartesian( float[], float[] );
> methods can inform the calling routine that it was unable to calculate a value, what are your
> thoughts on this use-case? Cheers Douglas > On 11/06/2015, at 11:05 am, Douglas Pearless
> <Douglas.Pearless@pearless.co.nz> wrote: > > I was under the assumption that not changing the
> actuator mm in this routine meant that the calling routine will receive back whatever the actuator
> mm was prior to calling, i.e. no change if the code cannot return a valid value? > >> On 11/06/2015,
> at 10:45 am, Jim Morris <notifications@github.com <mailto:notifications@github.com>> wrote: > > In
> src/modules/robot/Robot.cpp
> <https://github.com/Smoothieware/Smoothieware/pull/673#discussion_r32174728>: > > > @@ -195,20
> +201,20 @@ void Robot::on_config_reload(void *argument) > > Pin gamma_dir_pin; > > Pin gamma_en_pin;
>  > > > > - alpha_step_pin.from_string( THEKERNEL->config->value(alpha_step_pin_checksum
> )->by_default("2.0" )->as_string())->as_output(); > > - alpha_dir_pin.from_string(
> THEKERNEL->config->value(alpha_dir_pin_checksum )->by_default("0.5" )->as_string())->as_output(); >
>  > - alpha_en_pin.from_string( THEKERNEL->config->value(alpha_en_pin_checksum )->by_default("0.4"
> )->as_string())->as_output(); > > - beta_step_pin.from_string(
> THEKERNEL->config->value(beta_step_pin_checksum )->by_default("2.1" )->as_string())->as_output(); >
>  > - beta_dir_pin.from_string( THEKERNEL->config->value(beta_dir_pin_checksum )->by_default("0.11"
> )->as_string())->as_output(); > > - beta_en_pin.from_string(
> THEKERNEL->config->value(beta_en_pin_checksum )->by_default("0.10" )->as_string())->as_output(); > >
> - gamma_step_pin.from_string( THEKERNEL->config->value(gamma_step_pin_checksum )->by_default("2.2"
> )->as_string())->as_output(); > > - gamma_dir_pin.from_string(
> THEKERNEL->config->value(gamma_dir_pin_checksum )->by_default("0.20" )->as_string())->as_output(); >
>  > - gamma_en_pin.from_string( THEKERNEL->config->value(gamma_en_pin_checksum )->by_default("0.19"
> )->as_string())->as_output(); > > + alpha_step_pin.from_string( THEKERNEL->config->value((unsigned
> int)alpha_step_pin_checksum )->by_default("2.0" )->as_string())->as_output(); > remove casts... > >
> â€” > Reply to this email directly or view it on GitHub
> <https://github.com/Smoothieware/Smoothieware/pull/673/files#r32174728>. >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Smoothieware/Smoothieware/pull/673/files#r32177908>.
>